azure_arc_k8s_jumpstart/docs/aro_terraform.md

```diff
@@ -0,0 +1,232 @@
+    AROCLUSTER=arcarodemo           # the name of your Openshift Container cluster
+    ARC=arcarodemo                  # the name of the Arc Kubernetes resource
+    ```
+  * Create a resource group
+    ```bash
+    az group create --name $RESOURCEGROUP --location $LOCATION
+    ```
+    The following example output shows the resource group created successfully:
+    ```bash
+    {
+      "id": "/subscriptions/<guid>/resourceGroups/arcarodemo",
+      "location": "eastus",
+      "managedBy": null,
+      "name": "arcarodemo",
+      "properties": {
+          "provisioningState": "Succeeded"
+      },
+      "tags": null
+    }
+    ```
+
+  * **Create a virtual network**<br>
+    Azure Red Hat OpenShift clusters running OpenShift 4 require a virtual network with two empty subnets, for the master and worker nodes.
+
+    Create a new virtual network in the same resource group you created earlier.
+    ```bash
+    az network vnet create --resource-group $RESOURCEGROUP --name arcarodemo-vnet --address-prefixes 10.0.0.0/22
+    ```
+    The following example output shows the virtual network created successfully:
+    ```bash
+    {
+      "newVNet": {
+          "addressSpace": {
+          "addressPrefixes": [
+              "10.0.0.0/22"
+          ]
+          },
+          "id": "/subscriptions/<guid>/resourceGroups/arcarodemo/providers/Microsoft.Network/virtualNetworks/arcarodemo-vnet",
+          "location": "eastus",
+          "name": "arcarodemo-vnet",
+          "provisioningState": "Succeeded",
+          "resourceGroup": "arcarodemo",
+          "type": "Microsoft.Network/virtualNetworks"
+      }
+    }
+    ```
+  * Add an empty subnet for the master nodes.
+    ```bash
+    az network vnet subnet create \
+    --resource-group $RESOURCEGROUP \
+    --vnet-name arcarodemo-vnet \
+    --name master-subnet \
+    --address-prefixes 10.0.0.0/23 \
+    --service-endpoints Microsoft.ContainerRegistry
+    ```
+
+  * Add an empty subnet for the worker nodes.
+    ```bash
+    az network vnet subnet create \
+    --resource-group $RESOURCEGROUP \
+    --vnet-name arcarodemo-vnet \
+    --name worker-subnet \
+    --address-prefixes 10.0.2.0/23 \
+    --service-endpoints Microsoft.ContainerRegistry
+    ```
+  * [Disable subnet private endpoint policies](https://docs.microsoft.com/en-us/azure/private-link/disable-private-link-service-network-policy) on the master subnet. This is required to be able to connect and manage the cluster.
+    ```bash
+    az network vnet subnet update \
+    --name master-subnet \
+    --resource-group $RESOURCEGROUP \
+    --vnet-name arcarodemo-vnet \
+    --disable-private-link-service-network-policies true
+    ```
+  * Create the Openshift Container Cluster:
+    ```bash
+    az aro create \
+    --resource-group $RESOURCEGROUP \
+    --name $AROCLUSTER \
+    --vnet arcarodemo-vnet \
+    --master-subnet master-subnet \
+    --worker-subnet worker-subnet
+    ```
```

Comment:
It feels that this can be more automated with ARM template and/or shell script with some env vars no?

Reply:
no it cannot since ARO doesn't have an ARM template and the steps to deploy it via ARM is going to include: 
1) Fetch the file 2) Change parameters File 3) run the az deployment command. instead of 1) Create a Vnet 2) Create subnet_One 3) create subnet_two 4) disable link az command

The flow for a user is better to just copy paste those commands in cloud cloud shell do the first set of outlined steps.